### PR TITLE
feat: Add base config for JS loader with performance/replay

### LINKED ISSUE #46368: JS Loader: Sentry.init should be called - with defaults from docs for Performance/Replay
As the title says, the goal should be that someone should be able to just paste the `<script/>` tag and without any manual calls to `Sentry.init` get the experience using it.
That means, for example, pageload transactions and Replays are automatically captured when Performance and/or Replay is enabled.

It should be possible to still call `Sentry.init` yourself and with that overwrite our defaults.

### PR DESCRIPTION
**Pull request by mydea**
This PR adjusts the loader to include default config when enabling debug/performance/replay.

We should probably do a bit more testing etc. for this, as this is not really covered well as of now.
I did some tests locally with different scenarios, where stuff worked well, as far as I can tell.

## Lazy mode

Lazy mode is only available for errors-only config. Users can manually opt-out of this by adding `data-lazy="no"` to their script tag.

In lazy mode, nothing happens until an error occurs. Once an error occurs, we download the CDN bundle, and then...:

* If `onLoad` is configured, trigger it
* If `SDK.init()` has not been called, call it with defaults

## Eager mode

If either performance or replay is enabled, we are always in eager (non-lazy) mode. In this mode, we'll automatically load the CDN bundle, and then...:

* If `onLoad` is configured, trigger it
* If `SDK.init()` has not been called, call it with defaults

## Defaults

The following default config is defined:

* `dsn`: Always set
* `debug: true` is set when debug bundles are enabled
* `tracesSampleRate: 0.1` is set when performance is enabled
* `replaysSessionSampleRate: 0.1` & `replaysOnErrorSampleRate: 1` are set when replay is enabled
* `integrations`: Will add `Sentry.BrowserTracing` and/or `Sentry.Replay` if performance/replay is enabled

This config can be overwritten by the user, and is merged. So e.g. if they do:

```js
Sentry.onLoad(function() {
  Sentry.init({
    tracesSampleRate: 0.5
  });
});
```

For a full loader build with every option enabled, the actual config will be:

```js
{
  dsn: 'your-dsn',
  debug: true,
  tracesSampleRate: 0.5,
  replaysSessionSampleRate: 0.1,
  replaysOnErrorSampleRate: 1,
  integrations: [new Sentry.BrowserTracing(), new Sentry.Replay()]
}
```

Closes https://github.com/getsentry/sentry/issues/46368

### CODE REVIEW

**Review by AbhiPrasad — APPROVED** [review]

**Comment by mydea** [comment]
So I updated this quite a bit:

* Wrote tests in https://github.com/getsentry/sentry-javascript/pull/7662 to actually test that the loader works with different combinations of lazy/products etc - this found some issues that we fixed in both repos 🎉 
* Enforce that when performance or replay is enabled, we use SDK v7. This is also enforced UI-side now:

image

When you switch your SDK version to <7, it automatically removes performance & tracing. This should ensure we do not get into a weird state.

* When Replay is enabled, we show a message that this will result in the ES6 bundle:

image

ref https://github.com/getsentry/sentry/issues/46436

File: static/app/views/settings/project/projectKeys/details/keySettings.tsx [hunk 1]
```diff
@@ -267,14 +301,34 @@ export function KeySettings({onRemove, organization, params, data}: Props) {
...
+                          !sdkVersionSupportsPerformanceAndReplay(browserSdkVersion))
+                      }
+                      help={
+                        value.requiresV7 &&
+                        !sdkVersionSupportsPerformanceAndReplay(browserSdkVersion)
+                          ? t('Only available in SDK version 7.x and above')
+                          : key === DynamicSDKLoaderOption.HAS_REPLAY &&
+                            dynamicSDKLoaderOptions[sdkLoaderOption]
+                          ? t(
+                              'When using Replay, the loader will load the ES6 instead of the ES5 bundle.'
```
Comment (mydea) [inline static/app/views/settings/project/projectKeys/details/keySettings.tsx:328]: Not so sure about the wording here... maybe:

> When using Replay, Sentry is not ES5 compatible.

Or 

> When using Replay, we use the ES6 bundle of Sentry.

**Review by AbhiPrasad — CHANGES_REQUESTED** [review]
We'll need to merge in the backend changes first - and then do the frontend.

This is still fine since everything is gated behind a feature flag.

**Review by AbhiPrasad — APPROVED** [review]

**Comment by mydea** [comment]
So I don't _think_ the getsentry test failures are related to this PR...?

File: src/sentry/web/frontend/js_sdk_loader.py [hunk 2]
```diff
@@ -108,9 +122,21 @@ def _get_context(
         except TypeError:
             sdk_url = ""  # It fails if it cannot inject the version in the string
 
+        config: SdkConfig = {"dsn": key.dsn_public}
+
+        if has_debug:
+            config["debug"] = True
+
+        if has_performance:
+            config["tracesSampleRate"] = 0.1
```
Comment (HazAT) [inline src/sentry/web/frontend/js_sdk_loader.py:131]:
```suggestion
            config["tracesSampleRate"] = 1
```
Let's change this to `1`

**Review by HazAT — APPROVED** [review]
Let's change the tracesSampleRate to `1`. The reason, it's a weird experience if you expect a transaction to show up when testing, but it doesn't.

**Pull request merged**